### PR TITLE
Debian package

### PR DESCRIPTION
**Pull request by sharkAndshark**
Add deb package  by [cargo-deb](https://crates.io/crates/cargo-deb)
- [ ] add deb package metadata
- [ ] more research on systemd integration

To inspect the package:
```shell
# install cargo-deb dependices
sudo apt install dpkg
sudo apt install dpkg-dev
sudo apt install liblzma-dev

cargo install cargo-deb
cd martin
cargo deb
The Debian package will be created in `target/debian/martin_0.8.7_amd64.deb` . 
```

To decompress this package:
```shell
dpkg -x martin_0.8.7_amd64.deb path_of_decompress  
```

### CODE REVIEW

**Comment by nyurik** [comment]
i'm all for adding .deb support, thx!

A few things:
* do we really need to set up readme param?  Per [docs](https://doc.rust-lang.org/cargo/reference/manifest.html#the-readme-field), readme param is already pointing to the readme file by default.
* does `[package.metadata.deb]` need to duplicate maintainer text, or can it simply reuse the main one?
* please add any needed targets to the `jsonfile`, probably similar to other ones that do a check first. Note that dependencies should be tested first, to avoid sudo without needing to do that.

**Comment by sharkAndshark** [comment]
Hi @nyurik Maybe we could keep these params?
## readme issue
Without `readme` param, the content of `readme` in the decompressed package would be:
```
../README.md
```

## 

## maintainer parame
Without maintainer in `[package.metadata.deb]`, there would be only one author based on the [cargo-deb doc](https://crates.io/crates/cargo-deb#packagemetadatadeb-options).
 `apt-cache show martin `   after `dpkg -i`  without maintainer in `[package.metadata.deb]` 
```shell
Package: martin
Status: install ok installed
Priority: optional
Installed-Size: 14998
Maintainer: Stepan Kuzmin <to.stepan.kuzmin@gmail.com>
......
```
`apt-cache show martin `    after `dpkg -i`  with maintainer in in `[package.metadata.deb]`
```shell
Package: martin
Status: install ok installed
Priority: optional
Installed-Size: 14998
Maintainer: Stepan Kuzmin <to.stepan.kuzmin@gmail.com>, Yuri Astrakhan <YuriAstrakhan@gmail.com>, MapLibre contributors
......
```
##  CI
You mentioned adding any necessary targets to the `jsonfile`, but I didn't find any such file in the repository. Did you mean to suggest updating the [CI workflow](https://github.com/maplibre/martin/blob/main/.github/workflows/ci.yml) to include publishing a deb package?

**Comment by nyurik** [comment]
@sharkAndshark I meant [this file](https://github.com/maplibre/martin/blob/main/justfile). Thx for explaining about the fields.

**Comment by sharkAndshark** [comment]
Hi @nyurik  :)
By now, we could generate debian package in `CI.yml` or `jsut debian-package`.   
But I think there are still some chaos in the code . Would you mind give some guide?

File: justfile [hunk 1]
```diff
@@ -118,6 +118,14 @@ mdbook:
     fi
     mdbook serve docs --open --port 8321
 
+# Build debian package
+deb-package:
```
Comment (nyurik) [inline justfile]: i think we may have other packages later, so perhaps this is better?

```suggestion
package-deb:
```

File: justfile [hunk 2]
```diff
@@ -118,6 +118,14 @@ mdbook:
...
+    @if ! command -v "cargo deb" &> /dev/null; then \
```
Comment (nyurik) [inline justfile]: there is no such thing as a `cargo deb` command - command (binary file) has no spaces.  When cargo extensions are installed, they all have `cargo-deb` and similar names.

```suggestion
    @if ! command -v "cargo-deb" &> /dev/null; then \
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -29,6 +29,8 @@ jobs:
         include:
           - os: ubuntu-latest
             target: x86_64-unknown-linux-gnu
+          - os: ubuntu-latest
```
Comment (nyurik) [inline .github/workflows/ci.yml]: Lets not create a new job run here - it requires a lot of CI resources. Instead, just add an additional deb build step or if-condition as part of the `ubuntu-latest / x86_64-unknown-linux-gnu`

**Review by nyurik — COMMENTED** [review]
thanks, looks really good, I think we are almost there, left a few comments

File: justfile (hunk 2 above)
Comment (nyurik) [reply justfile]: P.S. In #752 I have improved dependency installation, so you can simply do `package-deb: (cargo-install "cargo-deb")`. It should merge soon.

Comment (nyurik) [reply justfile]: Update: that PR has merged, you can now install `cargo-deb` with `package-deb: (cargo-install "cargo-deb")`

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -62,6 +62,13 @@ jobs:
         run: |
           echo "VCPKG_ROOT=$env:VCPKG_INSTALLATION_ROOT" | Out-File -FilePath $env:GITHUB_ENV -Append
           vcpkg install openssl:x64-windows-static-md
+      - name: Install cargo-deb
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt install dpkg -y
+          sudo apt install dpkg-dev -y
+          sudo apt install liblzma-dev -y
```
Comment (nyurik) [inline .github/workflows/ci.yml]:
```suggestion
          sudo apt install -y dpkg dpkg-dev liblzma-dev
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -156,7 +169,7 @@ jobs:
         with:
           name: build-${{ matrix.target }}
           path: target/
-      - name: Integration Tests
+      - name: Integration Tests (without debian package)
```
Comment (nyurik) [inline .github/workflows/ci.yml]: Not sure why change of the name?

```suggestion
      - name: Integration Tests
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -197,6 +221,13 @@ jobs:
             tar czvf ../${{ matrix.name }} martin${{ matrix.ext }} mbtiles${{ matrix.ext }}
           fi
           cd -
+      - name: package (debian)
```
Comment (nyurik) [inline .github/workflows/ci.yml]:
```suggestion
      - name: Package (debian)
```

**Review by nyurik — COMMENTED** [review]
almost there, a few tiny auto-fixes, and we can merge :)

**Review by nyurik — APPROVED** [review]

**Pull request merged**